### ISSUE TITLE
False positive ERA001 for #region foldouts in PyCharm

### ISSUE THREAD

**Issue by bshishov** [charliermarsh/ruff#4538]
Hello, 

there is a code folding functionality using comment lines `# region {region name}` in JetBrains IDEs. End ERA001 fires as there is code inside a region name. 

```python
# region: Foo / Bar
# endregion
# region: Not a code
# endregion
# region: 1 + 1
# endregion

class SomeName:
    pass

# region: SomeName
# endregion
```

```
ruff --isolated --select ERA folding.py
folding.py:1:1: ERA001 [*] Found commented-out code
folding.py:5:1: ERA001 [*] Found commented-out code
folding.py:11:1: ERA001 [*] Found commented-out code
```

Notice that code-like names are detected as rule violation and `# region: Not a code` not.

Ruff version is `0.0.269`

The best documentation I could find: 
* https://www.jetbrains.com/help/idea/code-folding-settings.html
* https://www.jetbrains.com/help/rider/Coding_Assistance__Surrounding_with_Region.html
This feature was implemented in JetBrains IDEs to mimic same feature in Visual Studio editor.

There is also some extensions for different IDEs like VSCode: https://marketplace.visualstudio.com/items?itemName=maptz.regionfolder and also a discussion https://github.com/microsoft/vscode/issues/105 

I couldn't find any implementation of the allowed pattern for start and end myself. My best guess for PyCharm is following patterns
* `#\s*region[:\s]+(?<name>.*)` for start of named regions
* `#\s*region.*` for start of unnamed regions
* `#.*endregion.*` for end of any regions
